app/generate.js

```diff
@@ -97,8 +99,24 @@ function generatePages(renderedComponents) {
       page.conclusionHtml = '';
     }
 
-    page.header = header;
-    page.footer = footer;
+    if (!page.header) {
+      page.header = defaultHeader;
+    } else {
+      if (!page.header.logo) {
+        throw new Error('Header must have logo');
+      }
+      page.header = headerTemplate({
+        url: page.header.logo.url,
+        description: page.header.logoDescription,
+        tagline: page.header.tagline,
+      });
```

Comment:
The above lines are (or should be) an exact repeat of the construction of the default header for the org. Please break this out into a separate function. Call it `generateHeader`